### PR TITLE
Add Tag filter

### PR DESCRIPTION
**Pull request by Bakual**
Pull Request to add tag filter option to article blog menu item
### Summary of Changes

Adds a new parameter to the menuitem for category article blogs. The necessary code for the actual filtering is already present in the model for quite some time.
### Testing Instructions

Create a new menu item for the com_content articles blog and select a tag in the menu item.
You should only get the articles with that selected tag.
### Documentation Changes Required

None

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on ff875dae384b91513f5f6dca87abfbe05811b083<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12640">issues.joomla.org/tracker/joomla-cms/12640</a>.</sub>

File: components/com_content/views/category/tmpl/blog.xml [hunk 1]
```diff
@@ -18,6 +18,12 @@
 				label="JGLOBAL_CHOOSE_CATEGORY_LABEL"
 				required="true"
 			/>
+			<field name="filter_tag" type="tag"
```
Comment (brianteeman) [inline components/com_content/views/category/tmpl/blog.xml]: can you put type="tag" on a new line please

Comment (Bakual) [inline components/com_content/views/category/tmpl/blog.xml]: Done

**Comment by brianteeman** [comment]
The code works perfectly - unfortunately the text for the tooltip is completely wrong

image

you will need to add a new string in en-GB.ini 

> JTAGS_FIELD_SELECT="Select the tag to use."

**Comment by brianteeman** [comment]
@bakual I did two PR on your branch to add the new string

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on ca1f53569236cf759e72f86e16199fdc3ad8d27b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12640">issues.joomla.org/tracker/joomla-cms/12640</a>.</sub>

**Comment by Bakual** [comment]
I had to manually do the one (was for wrong branch) and messed it up a bit. But now it should be fine.
Thanks for the input!

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 263ba57c1f0019333abe6142755af3f91b998b6e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12640">issues.joomla.org/tracker/joomla-cms/12640</a>.</sub>

**Comment by brianteeman** [comment]
I thought the same but its for another pr.

**Comment by jeckodevelopment** [comment]
I have tested this item :white_check_mark: successfully on 263ba57c1f0019333abe6142755af3f91b998b6e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12640">issues.joomla.org/tracker/joomla-cms/12640</a>.</sub>

**Comment by Bakual** [comment]
> Shouldn't we allow several tags, instead of just one?

The current code (https://github.com/joomla/joomla-cms/blob/staging/components/com_content/models/articles.php#L88-L89) only allows an integer. If you want multiple tags, you will have to adjust the filtering code first so it can take an array of ids (and for B/C an integer). Com_tags itself has several places where it accepts a single tag, a coma-separated list of tags or an array of tags. We had some not-so-funny security issues because of that.... So if you want to do it, make sure input filtering is done properly.

> Shouldn't we made the same funcionality to category list menu too?
> Shouldn't we made the same funcionality to all menu category in other components too (ex: list contacts in a category)?

If you think this is important, feel free to make a PR. I just did this to show it's already possible.

**Comment by andrepereiradasilva** [comment]
Sure. it can be other PR.

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12640">issues.joomla.org/tracker/joomla-cms/12640</a>.</sub>

**Pull request merged**